jgrapht-core/src/test/java/org/jgrapht/alg/spanning/MinimumSpanningTreeTest.java

```diff
@@ -17,100 +17,74 @@
+    public static final Integer H = "H".codePointAt(0);
 
     // ~ Instance fields --------------------------------------------------------
 
-    private DefaultWeightedEdge AB;
-    private DefaultWeightedEdge AC;
-    private DefaultWeightedEdge BD;
-    private DefaultWeightedEdge DE;
-    private DefaultWeightedEdge EG;
-    private DefaultWeightedEdge GH;
-    private DefaultWeightedEdge FH;
+    public static DefaultWeightedEdge AB;
+    public static DefaultWeightedEdge AC;
+    public static DefaultWeightedEdge BD;
+    public static DefaultWeightedEdge DE;
+    public static DefaultWeightedEdge EG;
+    public static DefaultWeightedEdge GH;
+    public static DefaultWeightedEdge FH;
 
     // ~ Methods ----------------------------------------------------------------
 
     @Test
-    public void testKruskal()
-    {
-
-        testMinimumSpanningTreeBuilding(
-            new KruskalMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleConnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, DE), 15.0);
-
-        testMinimumSpanningTreeBuilding(
-            new KruskalMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleDisconnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, EG, GH, FH), 60.0);
-    }
-
-    @Test
-    public void testPrim()
-    {
-        testMinimumSpanningTreeBuilding(
-            new PrimMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleConnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, DE), 15.0);
-
-        testMinimumSpanningTreeBuilding(
-            new PrimMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleDisconnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, EG, GH, FH), 60.0);
+    public void testSimpleDisconnectedWeightedGraph(){
+        testMinimumSpanningTreeBuilding(createSolver(createSimpleDisconnectedWeightedGraph()).getSpanningTree(),
+                Arrays.asList(AB, AC, BD, EG, GH, FH), 60.0);
     }
 
     @Test
-    public void testBoruvska()
-    {
-        testMinimumSpanningTreeBuilding(
-            new BoruvkaMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleConnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, DE), 15.0);
-
-        testMinimumSpanningTreeBuilding(
-            new BoruvkaMinimumSpanningTree<String, DefaultWeightedEdge>(
-                createSimpleDisconnectedWeightedGraph()).getSpanningTree(),
-            Arrays.asList(AB, AC, BD, EG, GH, FH), 60.0);
+    public void testSimpleConnectedWeightedGraph() {
+        testMinimumSpanningTreeBuilding(createSolver(createSimpleConnectedWeightedGraph()).getSpanningTree(),
+                Arrays.asList(AB, AC, BD, DE), 15.0);
     }
 
     @Test
-    public void testRandomInstances()
```

Comment:
This test is not a good idea. Here's what will happen.

PrimMinimumSpanningTreeDenseGraphsTest executes `super.testRandomInstances()`.
PrimMinimumSpanningTreeTest executes `super.testRandomInstances()`.
BoruvkaMinimumSpanningTreeTest executes `super.testRandomInstances()`.
KruskalMinimumSpanningTreeTest executes `super.testRandomInstances()`.

each time super.testRandomInstances() gets invoked, you compute:
1. `createSolver(g).getSpanningTree()`
2. `new KruskalMinimumSpanningTree`
3. `new PrimMinimumSpanningTree`
4. `new BoruvkaMinimumSpanningTree`

In other words, you are computing 4*4=16 times a spanning tree for the same graph. This leads to quite a bit of redundancy. Here's an easy fix:
```
SpanningTreeAlgorithm<.> solver=createSolver()
if(solver instanceOf BoruvkaMinimumSpanningTree)
  assertEquals(new Prim.getWeight(), solver.getTree.getWeight(), 1e-9);
else
  assertEquals(new BoruvkaMinimumSpanningTree.getWeight(), solver.getTree.getWeight(), 1e-9);
```

Here you simply ensure that you always compare the result against a tree produced by a different implementation: If Borukova was used, you compare against Prim, otherwise you can compare against Borukova.